.github/workflows/full-test.yml

```diff
@@ -32,6 +32,7 @@ jobs:
       run: |
         python -m pip install --upgrade pip
         python -m pip install pytest pytest-cov coveralls flake8
+        python -m pip install "numpy<2"
```

Comment:
This is a pretty strong constraint, do you know why this might be necessary?

Reply:
yes, the PyPI package for NEURON doesn't yet support NumPy v2

Reply:
I could test different versions of NumPy, but I'm guessing we'll be able to remove this constraint soon.